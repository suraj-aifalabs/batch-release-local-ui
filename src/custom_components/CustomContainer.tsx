import React, { ReactNode } from 'react';

interface CustomContainerProps {
    children: ReactNode;
}

const CustomContainer: React.FC<CustomContainerProps> = ({ children }) => {
    return (
        <div className='px-4'>{children}</div>
    );
};

export default CustomContainer;