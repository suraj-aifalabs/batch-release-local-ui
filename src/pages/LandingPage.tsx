import CustomButton from "@/custom_components/CustomButton"
import CustomContainer from "@/custom_components/CustomContainer"
import { Link } from "react-router-dom"
import { Input } from "@/components/ui/input"
import { useState } from "react"
const LandingPage = () => {
    const [username, setUsername] = useState("")
    const [email, setEmail] = useState("")
    return (
        <CustomContainer >
            <div className="flex flex-col justify-center h-[90dvh] gap-4 items-center w-full">
                <Input type="username" placeholder="Username" className="w-[300px]" value={username} onChange={(e) => setUsername(e.target.value)} />
                <Input type="email" placeholder="Email" className="w-[300px]" value={email} onChange={(e) => setEmail(e.target.value)} />
                <Link state={{ username, email }} to="/setting/release"><CustomButton type="primary" text="Login" /></Link>
            </div >
        </CustomContainer>
    )
}

export default LandingPage
