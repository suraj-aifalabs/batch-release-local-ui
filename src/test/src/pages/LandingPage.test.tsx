import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import LandingPage from '@/pages/LandingPage';
import CustomContainer from '@/custom_components/CustomContainer';
import { MemoryRouter } from 'react-router-dom';

jest.mock('@/custom_components/CustomButton', () => {
    return jest.fn(({ text, ...props }) => (
        <button {...props}>{text}</button>
    ));
});

jest.mock('@/custom_components/CustomContainer', () => {
    return jest.fn(({ children }) => (
        <div data-testid="custom-container">{children}</div>
    ));
});

describe('LandingPage component', () => {
    it('renders without crashing', () => {
        render(
            <MemoryRouter>
                <LandingPage />
            </MemoryRouter>
        );
        expect(screen.getByTestId('custom-container')).toBeInTheDocument();
    });



    it('wraps content in CustomContainer', () => {
        render(
            <MemoryRouter>
                <LandingPage />
            </MemoryRouter>
        );

        expect(CustomContainer).toHaveBeenCalled();
        const container = screen.getByTestId('custom-container');
        expect(container).toBeInTheDocument();
    });

    it('has correct link navigation', () => {
        render(
            <MemoryRouter>
                <LandingPage />
            </MemoryRouter>
        );

        const link = screen.getByRole('link');
        expect(link).toHaveAttribute('href', '/setting/release');
    });

    it('applies correct layout styles', () => {
        render(
            <MemoryRouter>
                <LandingPage />
            </MemoryRouter>
        );

        const innerDiv = screen.getByTestId('custom-container').firstChild;
        expect(innerDiv).toHaveClass('flex');
        expect(innerDiv).toHaveClass('flex-col');
        expect(innerDiv).toHaveClass('justify-center');
        expect(innerDiv).toHaveClass('items-center');
        expect(innerDiv).toHaveClass('h-[90dvh]');
        expect(innerDiv).toHaveClass('gap-4');
        expect(innerDiv).toHaveClass('w-full');
    });

    it('matches snapshot', () => {
        const { asFragment } = render(
            <MemoryRouter>
                <LandingPage />
            </MemoryRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});