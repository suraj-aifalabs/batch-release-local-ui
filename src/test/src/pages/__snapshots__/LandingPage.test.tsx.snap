// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage component matches snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="custom-container"
  >
    <div
      class="flex flex-col justify-center h-[90dvh] gap-4 items-center w-full"
    >
      <a
        data-discover="true"
        href="/setting/release"
      >
        <button
          class="w-full"
          type="primary"
        >
          View PDF
        </button>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
